examples/controlnet/train_controlnet.py

```diff
@@ -604,9 +605,22 @@ def tokenize_captions(examples, is_train=True):
         )
         return inputs.input_ids
 
+    def resize_multiple_8(image):
+        height = image.height
+        width = image.width
+
+        # round down to nearest multiple of 8
+        height = (height // 8) * 8
+        width = (width // 8) * 8
+
+        image = TF.resize(image, (height, width), transforms.InterpolationMode.BILINEAR)
+
+        return image
+
     image_transforms = transforms.Compose(
         [
             transforms.Resize(args.resolution, interpolation=transforms.InterpolationMode.BILINEAR),
+            resize_multiple_8,
```

Comment:
Couldn't we do both at the same time? Two consecutive bilinear interpolations is not ideal imo.

Reply:
Yeah that's what I first wanted to do but that requires implementing the logic for determining the proper aspect ratio for the single resize :P I think it sounds like doing square images via crops might be the better solution